sql/core/src/test/resources/sql-tests/inputs/interval.sql

```diff
@@ -97,17 +97,24 @@ select interval 1 year 2 month 3 week 4 day 5 hour 6 minute 7 seconds 8 millisec
 select interval '30' year '25' month '-100' day '40' hour '80' minute '299.889987299' second;
 select interval '0 0:0:0.1' day to second;
 select interval '10-9' year to month;
+-- SPARK-29933: ThriftServerQueryTestSuite runs tests with wrong settings
+set spark.sql.dialect=Spark;
```

Comment:
hmm, why we set the config to its default value?

Reply:
Look at the comment above, `ThriftServerQueryTestSuite` runs this in the PostgreSQL dialect, and `select interval '20 15' day to hour` fails because current implementation (enabled for PostgreSQL dialect) has the bug: https://github.com/apache/spark/pull/26473#issuecomment-554456769

Reply:
Even the explicit set to the Spark dialect does not work: https://github.com/apache/spark/pull/26473#issuecomment-556135535 . This PR is blocked by a bug in thrift server itself or in `ThriftServerQueryTestSuite`. When I run the tests via `build/sbt "testOnly *SQLQueryTestSuite -- -z interval.sql"`, everything is ok:
```
[info] SQLQueryTestSuite:
[info] - interval.sql (15 seconds, 155 milliseconds)
[info] - postgreSQL/interval.sql (1 second, 809 milliseconds)
[info] - ansi/interval.sql (7 seconds, 654 milliseconds)
[info] Run completed in 33 seconds, 159 milliseconds.
[info] Total number of tests run: 3
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 3, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```